pkg/server/handler_test.go

```diff
@@ -153,3 +164,104 @@ func TestProcessResourceIdentity(t *testing.T) {
+			path:                "/api/v1/namespaces/default/pods/foo/exec",
+			workspace:           "cluster1",
+			podExists:           true,
+			syncTargetExists:    true,
+			synctargetWorkspace: "cluster2",
+			expectedPath:        "/services/syncer-tunnels/clusters/cluster2/apis/workload.kcp.dev/v1alpha1/synctargets/synctarget1/proxy/api/v1/namespaces/kcp-1pg4utc35m9j/pods/foo/exec",
+		},
+		"non existing pod": {
+			path:        "/api/v1/namespaces/default/pods/foo/exec",
+			workspace:   "cluster1",
+			podExists:   false,
+			expectError: true,
+		},
+		"non existing synctarget": {
+			path:             "/api/v1/namespaces/default/pods/foo/exec",
+			workspace:        "cluster1",
+			podExists:        true,
+			syncTargetExists: false,
+			expectError:      true,
+		},
+		"invalid request": {
+			path:        "/api/v1/namespaces/default/pods/foo/invalid",
+			workspace:   "cluster1",
+			podExists:   true,
+			expectError: true,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+			podSubresourceProxyHandlerBuilder := &podSubresourceProxyHandlerBuilder{
+				apiHandler:              http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}),
+				podSubresourcePathRegex: regexp.MustCompile(`/api/v1/namespaces/(?P<Namespace>[^/]+)/pods/(?P<PodName>[^/]+)/(?P<Subresource>exec|log|portforward|proxy|attach)?(?P<Arguments>.*)`),
+				getPodByName: func(ctx context.Context, cluster logicalcluster.Name, namespace, podName string) (*corev1.Pod, error) {
+					if !tc.podExists {
+						return nil, errors.NewNotFound(schema.GroupResource{Resource: "pods"}, podName)
+					}
+					return &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      podName,
+							Namespace: namespace,
+							Labels: map[string]string{
+								"state.workload.kcp.dev/ABCDEFGHIJKL": "Upsync",
+							},
+						},
+					}, nil
+				},
+				getSyncTargetBySynctargetKey: func(ctx context.Context, synctargetKey string) (*workloadv1alpha1.SyncTarget, error) {
+					if !tc.syncTargetExists {
+						return nil, errors.NewNotFound(schema.GroupResource{Resource: "synctargets"}, synctargetKey)
+					}
+					return &workloadv1alpha1.SyncTarget{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "synctarget1",
+							Annotations: map[string]string{
+								"workload.kcp.dev/key": "ABCDEFGHIJKL",
+								"kcp.dev/cluster":      tc.synctargetWorkspace,
+							},
+						},
+					}, nil
+				},
+			}
+			r := httptest.NewRequest("GET", tc.path, nil).WithContext(request.WithCluster(ctx, request.Cluster{Name: logicalcluster.New(tc.workspace)}))
+			rw := httptest.NewRecorder()
+
+			handler := podSubresourceProxyHandlerBuilder.build()
+			handler.ServeHTTP(rw, r)
+			result := rw.Result()
+			bytes, err := io.ReadAll(result.Body)
+			require.NoError(t, err, "Request body cannot be read")
+			if tc.expectError {
```

Comment:
Don't you want to test the detail of the error, with something like `expectedError string` ?

Reply:
done